Gordon360/ApiControllers/JobsController.cs

```diff
@@ -0,0 +1,223 @@
+            userID = Convert.ToInt32(id);
+
+
+            string query = "SELECT * from student_timesheets where ID_NUM = " + userID + ";";
+            IEnumerable<StudentTimesheetsViewModel> queryResult = null;
+
+            try
+            {
+                queryResult = RawSqlQuery<StudentTimesheetsViewModel>.StudentTimesheetQuery(query);
+            }
+            catch (Exception e)
+            {
+                //
+                System.Diagnostics.Debug.WriteLine(e.Message);
+            }
+            return Ok(queryResult);
+        }
+
+        /// <summary>
+        /// Get a user's active jobs
+        /// </summary>
+        /// <param name="shiftDetails"></param>
+        /// <returns>The user's active jobs</returns>
+        [HttpPost]
+        [Route("saveShift")]
+        [StateYourBusiness(operation = Operation.ADD, resource = Resource.SHIFT)]
+        public HttpResponseMessage saveShiftForUser([FromBody] ShiftViewModel shiftDetails)
+        {
+            IEnumerable<StudentTimesheetsViewModel> result = null;
+            IEnumerable<OverlappingShiftIdViewModel> overlapCheckResult = null;
+
+            int userID = -1;
+
+            var authenticatedUser = this.ActionContext.RequestContext.Principal as ClaimsPrincipal;
+            var username = authenticatedUser.Claims.FirstOrDefault(x => x.Type == "user_name").Value;
+            var id = _accountService.GetAccountByUsername(username).GordonID;
+            userID = Convert.ToInt32(id);
+
+            try
+            {
+                overlapCheckResult = _jobsService.checkForOverlappingShift(userID, shiftDetails.SHIFT_START_DATETIME, shiftDetails.SHIFT_END_DATETIME);
+                System.Diagnostics.Debug.WriteLine("overlap check result: " + overlapCheckResult.Count());
+                if (overlapCheckResult.Count() > 0)
+                {
+                    return Request.CreateResponse(HttpStatusCode.Conflict, "Error: shift overlap detected");
+                }
+                result = _jobsService.saveShiftForUser(userID, shiftDetails.EML, shiftDetails.SHIFT_START_DATETIME, shiftDetails.SHIFT_END_DATETIME, shiftDetails.HOURS_WORKED, shiftDetails.SHIFT_NOTES, username);
+            }
+            catch (Exception e)
+            {
+                System.Diagnostics.Debug.WriteLine(e.Message);
+                return Request.CreateResponse(HttpStatusCode.InternalServerError, e);
+            }
+            return Request.CreateResponse(HttpStatusCode.OK, result);
+        }
+
+        /// <summary>
+        /// Get a user's active jobs
+        /// </summary>
+        /// <returns>The result of deleting the shift</returns>
+        [HttpDelete]
+        [Route("deleteShift/{rowID}")]
+        [StateYourBusiness(operation = Operation.DELETE, resource = Resource.SHIFT)]
+        public IHttpActionResult deleteShiftForUser(int rowID)
+        {
+            IEnumerable<StudentTimesheetsViewModel> result = null;
+            int userID = -1;
```

Comment:
could 154-159 made into a function of their own. It looks like you reuse that code a lot

Reply:
though, looks like there are some variables you set within those lines that you use besides userID, which seemed to be the final result